### PR TITLE
feat: update-country implemented

### PR DESCRIPTION
**Pull request by obajide028**
# Pull Request Description

Please describe the changes you are making in this PR and the reason for them.

- [ ] Bug Fix
- [x] New Feature
- [ ] Code Refactor
- [ ] Documentation Update

### Description of the changes:

---

## Checklist

Before submitting your pull request, please ensure the following:

- [x] I have added unit and integration tests with complete code coverage(for new features) to prove my feature works.
- [ ] I have updated unit or integration tests(for bug fix or code refactor) for my fix.
- [ ] I have updated relevant documentation (if applicable).
- [x] I did not skip git pre commit hooks (Required).

---

### Additional Notes

Tick the boxes above that matches what you have done. Thank you. Happy Pull Request.

### CODE REVIEW

File: src/controllers/country/update-country.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  try {
+    const validatedId = validator.escape(id.trim());
+    const updatedCountry = await updateCountryService(validatedId, name);
+    if (!updatedCountry) {
+      res.status(404).json({ message: COUNTRY_NOT_FOUND });
+      return;
+    }
+
+    res.status(200).json({ message: SUCCESS_MESSAGE, data: updatedCountry });
```
Comment (godwintrav) [inline src/controllers/country/update-country.ts:34]: You forgot to return from the function here and inside the catch block

File: src/services/__tests__/country-service.test.ts [hunk 2]
```diff
@@ -56,3 +57,30 @@ describe('deleteCountryService', () => {
...
+// Test update a country
+describe('updateCountryService', () => {
+  it('should update and return the updated country', async () => {
+    const mockCountry = { _id: '123', name: 'naija' };
+
+    // mock the find and update country method to return mock countries
+    (Country.findByIdAndUpdate as jest.Mock).mockResolvedValue(mockCountry);
+
+    //call the service
+    const result = await updateCountryService('123', 'naija');
```
Comment (godwintrav) [inline src/services/__tests__/country-service.test.ts]: Use the mockCountry object you created for the function parameters to follow DRY principle. Call mockCountry._id or mockCountry.name

File: src/services/__tests__/country-service.test.ts [hunk 3]
```diff
@@ -56,3 +57,30 @@ describe('deleteCountryService', () => {
...
+    expect(Country.findByIdAndUpdate).toHaveBeenCalledWith(
+      '123',
```
Comment (godwintrav) [inline src/services/__tests__/country-service.test.ts]: Same here

File: src/services/__tests__/country-service.test.ts (hunk 2 above)
Comment (obajide028) [reply src/services/__tests__/country-service.test.ts]: Okay boss

File: src/controllers/country/update-country.ts (hunk 1 above)
Comment (obajide028) [reply src/controllers/country/update-country.ts:34]: Okay boss

File: src/controllers/country/update-country.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  const { name } = req.body;
+
+  if (!name) {
+    res.status(400).json({ message: NAME_REQUIRED });
+    return;
+  }
+
+  try {
+    const validatedId = validator.escape(id.trim());
+    const updatedCountry = await updateCountryService(validatedId, name);
```
Comment (godwintrav) [inline src/controllers/country/update-country.ts]: We will need to sanitise the name value sent to the API

**Review by godwintrav — APPROVED** [review]

**Pull request merged**